presto-spark-base/src/main/java/com/facebook/presto/spark/planner/IterativePlanFragmenter.java

```diff
@@ -265,9 +265,13 @@ public PlanNode visitExchange(ExchangeNode node, RewriteContext<FragmentProperti
         public PlanNode visitRemoteSource(RemoteSourceNode node, RewriteContext<FragmentProperties> context)
         {
             List<SubPlan> childSubPlans = node.getSourceFragmentIds().stream()
-                    .map(id -> subPlanByFragmentId.get(id))
+                    .map(subPlanByFragmentId::get)
                     .collect(toImmutableList());
             context.get().addChildren(childSubPlans);
+
+            // the partitioning handle should be the same as the handle from the partitioning scheme
+            // of any of the input fragments.
+            setDistributionForExchange(node.getExchangeType(), childSubPlans.get(0).getFragment().getPartitioningScheme(), context);
```

Comment:
Should we check that the partitioning scheme of the inputs are all the same or that's redundant (and maybe true by construction?)

Reply:
i think it's true by construction. they will be slightly different since they will have different output variables, but the partitioning handles will all be the same. (As I understand it fragment.partitioningScheme defines the output partitioning for the fragment and fragment.partitioning tells how the fragment itself will be distributed.  The output partitioning of a childFragment will necessarily match the partitioning of the parent fragment)